ocis 3.1 upgrade (#576)
owncloud/docs-ocis: modules/ROOT/pages/migration/upgrading-ocis.adoc

NOTE: For any Infinite Scale version to upgrade to, see {ocis-downloadpage-url}?sort=time&order=desc[download.owncloud.com,window=_blank] or {docker-ocis-url}[Infinite Scale image] to get the right version.

IMPORTANT: Before starting any upgrade, make a xref:maintenance/b-r/backup.adoc[backup] first.

IMPORTANT: If not explicitly mentioned, any upgrade is forward only and a backstep is neither supported nor recommended and will most likely break the instance.

> IMPORTANT: When upgrading from the current running release to the desired one, *ALL* upgrade steps from each release must be taken in order and none can be skipped.  

Reviewer: ```suggestion
IMPORTANT: When upgrading from an older release to the desired one, *ALL* upgrade steps from each release must be taken in order and none can be skipped.  
```

NOTE: For any Infinite Scale version to upgrade to, see {ocis-downloadpage-url}?sort=time&order=desc[download.owncloud.com,window=_blank] or {docker-ocis-url}[Infinite Scale image] to get the right version.

IMPORTANT: Before starting any upgrade, make a xref:maintenance/b-r/backup.adoc[backup] first.

IMPORTANT: If not explicitly mentioned, any upgrade is forward only and a backstep is neither supported nor recommended and will most likely break the instance.

> IMPORTANT: When upgrading from the current running release to the desired one, *ALL* upgrade steps from each release must be taken in order and none must be skipped.  

Reviewer 2: ```suggestion
IMPORTANT: When upgrading from the current running release to the desired one, *ALL* upgrade steps from each release must be taken in order and none can be skipped.  
```

== Version 3.0.0 to 3.1.0

=== Notable Changes Requiring Manual Intervention

> * The space index of the decomposedFS will be automatically migrated from symlinks to messagepack. This migration needs preperation to run successfully.

Reviewer: ```suggestion
* The space index of the decomposedFS will be automatically migrated from symlinks to messagepack. This migration needs preparation to run successfully.
```

* The xref:deployment/container/orchestration/orchestration.adoc#using-our-helm-charts-with-infinite-scale[Helm Chart] has been upgraded.

=== Upgrade Steps

. Shut down the Infinite Scale instance.
. Update Infinite Scale via:
+
--
* Docker
+
[source,bash]
----
docker pull owncloud/ocis:3.1.0
----

* or get the binary from the link above.
--

. Migrate the Space Index to Messagepack: +
It is important to disable clients and users logging in while the space index migration to `messagepack` is in progress. To do so, start Infinite Scale with only two necessary services. The example uses the binary as base:
+
--
[source,bash]
----
OCIS_RUN_SERVICES="storage-users,nats" \
ocis server
----

Because the proxy service is not running, no clients or users can login. During the restart of Infinite Scale, the symlinks of spaces in the DecomposedFS will be migrated to `messagepack` _automatically_ via the `storage-users` service. The start of the migration can be identified in the log by a message like:

[source,plaintext]
----

> {"level":"info","root":"/var/lib/ocis/storage/users","time":"2023-06-16T12:21:49.441395153Z","caller":"github.com/cs3org/reva/v2@v2.14.0/pkg/storage/utils/decomposedfs/migrator/0003_switch_to_messagepack_metadata.go:45","message":"Migrating to messagepack metadata backend..."}

Reviewer 3: Here is the new log:
```bash
{"level":"info","root":"/var/lib/ocis/storage/users","time":"2023-08-02T06:08:21.002825802Z","caller":"github.com/cs3org/reva/v2@v2.15.0/pkg/storage/utils/decomposedfs/migrator/0004_switch_to_messagepack_space_index.go:46","message":"Migrating /var/lib/ocis/storage/users/indexes/by-user-id/b7a2149b-3320-43a3-9bd2-c216340c212d.mpk to messagepack index format..."}
...
{"level":"info","root":"/var/lib/ocis/storage/users","time":"2023-08-02T06:08:21.028153362Z","caller":"github.com/cs3org/reva/v2@v2.15.0/pkg/storage/utils/decomposedfs/migrator/0005_fix_messagepack_space_index_format.go:37","message":"Fixing index format of /var/lib/ocis/storage/users/indexes/by-user-id/b7a2149b-3320-43a3-9bd2-c216340c212d.mpk"}
```

----

> When migration has finished - which can take a while depending on the quantity of files - you can see a log message like below:

Reviewer: ```suggestion
When migration has finished - which can take a while depending on the quantity of files - you can see a log message like this:
```

[source,plaintext]
----

> {"level":"info","time":"2023-06-16T12:30:05.267986656Z","caller":"github.com/cs3org/reva/v2@v2.14.0/pkg/storage/utils/decomposedfs/migrator/0003_switch_to_messagepack_metadata.go:106","message":"done."}`

Reviewer 3: Here is the new log:
```bash
{"level":"info","time":"2023-08-02T06:08:21.026943689Z","caller":"github.com/cs3org/reva/v2@v2.15.0/pkg/storage/utils/decomposedfs/migrator/0004_switch_to_messagepack_space_index.go:92","message":"done."}
...
{"level":"info","time":"2023-08-02T06:08:21.028422073Z","caller":"github.com/cs3org/reva/v2@v2.15.0/pkg/storage/utils/decomposedfs/migrator/0005_fix_messagepack_space_index_format.go:66","message":"done."}
```

Author: integrated
